src/locationsettings.cpp

```diff
@@ -466,6 +479,12 @@ bool LocationSettings::locationEnabled() const
 void LocationSettings::setLocationEnabled(bool enabled)
 {
     Q_D(LocationSettings);
+
+    if (d->isEmergencyCallActive()) {
```

Comment:
Pondering should it be allowed to turn the location/gps on.

Reply:
Why? AML does it already.

Reply:
And AML controls the location anyways, when battery is low it keeps running but shuts down all power consumers to have some grace period for the user to find a charger. Anyways, it would compete with AML requirements and the AML service, which will force the value to the file causing maybe confusion as in such case user would see "-> on -> off" and vice versa. Better to have this ignore all cases. 

Or maybe if we could have an additional QMceCallState or variable `aml_enabled` ? In such case all actions would be blocked but when AML is not enabled user could control this.

Reply:
Thinking of robustness and limiting only the minimum. AML is sort of an add-on so these rules on this level are not totally unproblematic, and also it's now assuming that AML will not use this component while on the other hand it would be good if all access to the file would go via this.

Reply:
AML is a service that at some point should also be launched at the lock UI so I think this is not available then. It is simple enough just to prevent user access to the options as there is a legislative requirement to do so. I left opening for enable only option in 517f7bd029dc2e9cc22e1b2323b63db32d0a3659 which might be needed. Anyways, all emergency calls != AML is on.

Reply:
I'm starting to think that if the situation is not yet clear enough to have a future-proof mw policy it might be better not to implement such a thing. Especially if the only gotcha would be having device user repeatedly and persistently toggling things in settings app during an ongoing emergency call. Or was there something else too?